src/controllers/user-controller.js

```diff
@@ -15,3 +17,19 @@ export const findAllUsers = (obj, args, { models }) => {
 export const findUserById = (_, { id }, { models }) => {
   return models.user.findByPk(id)
 }
+
+export const signUp = async (_, data, { models }) => {
+  const user = await models.user.findOne({ where: { username: input.username }}) 
+  if (!user){
+    user = await models.user.createUser({ data })
+    const secret = process.env.JWT_SECRET;
+    const token = jwt.sign({ sub: user.id }, secret, { exp: '10d' } );
+    return {
+      token,
+      user,
+    }
+  }
+  else {
+    throw new Error('Usuario ya existente');
```

Comment:
Fail loud, fail early.

This should be much clearer if you do it in the first line:
```javascript
if (user) throw new Error(...)

// the rest of your code here
```